src/ChannelImpl.h

```diff
@@ -150,6 +153,7 @@ class ChannelImpl {
   const char* name_;          ///< See GetName().
   int capacity_;              ///< See GetCapacity().
   int packet_size_;           ///< See GetPacketSize().
+  std::atomic_int dropped_packets_number_; ///< See GetDroppedPacketsNumber().
```

Comment:
Should I use the spin lock instead of atomic?

Reply:
It seems to me that the way this resource could be contested would be when multiple threads ran a single executor function concurrently, and tried to acquire a packet from the next channel in the pipeline at the same time, or if multiple nodes were trying to acquire a packet from the same node at the same time. I don't see how using the spinlock would make the necessary synchronization any better, but it could introduce further delay in processing the executor function of the node that tried to use this resource.

This is because threads in executor functions can't meaningfully proceed until the acquire call they are trying to make are completed, so potentially introducing delay here might not be what we want. We're only trying to access a small member variable for an increment, anyways, so this is probably not what the spinlocks were meant for (but rather, waiting on larger operations without wasting cycles and slices).

Can someone confirm this, or am I missing something?

Reply:
Actually I meant that maybe the code where I'm working with ```dropped_packets_number_``` could be restructured that it will be under a guarded code snippet, but yes, you are right, because maybe we should use the guard in the getter method too.

Reply:
The TryDroppingPacket() is not guarded, but Release() is actually guarded with the free queue's mutex. But because here, the snychronization is rather simple, we just increment value and read it out somewhere else, the atomic is good enough.

Reply:
I have one more remark: I understand why we need this for testing purposes. But a lot of very similar indicators will be logged as part of the self-profiling functionality. There, all synchronization issues will be generally solved anyway. So implementing this custom thing seems a bit redundant. Isn't it possible that we skip this atomic counter and instead we use something from the profiling? WDYT?